pkg/gadgets/trace/tcpconnect/tracer/bpf/tcpconnect.bpf.c

```diff
@@ -240,4 +328,16 @@ int BPF_KRETPROBE(ig_tcpc_v6_co_x, int ret)
 	return exit_tcp_connect(ctx, ret, 6);
 }
 
+SEC("kprobe/tcp_rcv_state_process")
+int BPF_KPROBE(ig_tcp_rsp, struct sock *sk)
+{
+	return handle_tcp_rcv_state_process(ctx, sk);
+}
+
+SEC("tracepoint/tcp/tcp_destroy_sock")
+int ig_tcp_destroy(struct trace_event_raw_tcp_event_sk *ctx)
+{
+	return cleanup_sock(ctx->skaddr);
```

Comment:
You can add a comment to say the tracepoint is fired for both IPv4 and IPv6.

Reply:
Is there a case where the tracepoint is called but handle_tcp_rcv_state_process has not been called? For a connection timeout? In that case, shouldn't we send an event to userspace?

Reply:
I agree that we should not do a big rework of this gadget logic in this PR.